pkgs/games/sil-q/default.nix

```diff
@@ -0,0 +1,65 @@
+{ lib, stdenv, fetchFromGitHub, writeScript, makeWrapper, ncurses, libX11 }:
+
+let
+  version = "v1.5.0";
+  src = fetchFromGitHub {
+    owner = "sil-quirk";
+    repo = "sil-q";
+    rev = version;
+    sha256 = "sha256-v/sWhPWF9cCKD8N0RHpwzChMM1t9G2yrMDmi1cZxdOs=";
+  };
+  setup = writeScript "setup" ''
+    mkdir -p "$ANGBAND_PATH"
+    # Copy all the data files into place.
+    cp -ar $1/* "$ANGBAND_PATH"
+    # The copied files are not writable, make them so.
+    chmod +w -R "$ANGBAND_PATH"
+  '';
+in stdenv.mkDerivation {
+  pname = "sil-q";
+  inherit version src;
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ ncurses libX11 ];
+
+  # Makefile(s) and config are not top-level.
+  sourceRoot = "source/src";
+
+  prePatch = ''
+    # Allow usage of ANGBAND_PATH.
+    substituteInPlace config.h --replace "#define FIXED_PATHS" ""
+
+    # Makefile.std has to be adapted in order to use ncurses.
+    # It says so in its own comments.
+    substituteInPlace Makefile.std --replace "-lcurses" "-lncurses"
+  '';
+
+  makefile = "Makefile.std";
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp sil $out/bin
+    wrapProgram $out/bin/sil \
+      --run "set -u" \
+      --run "export ANGBAND_PATH=\$HOME/.sil" \
```

Comment:
Pleas use the flag to set environment variables instead of shelling.

Reply:
`--set` was what I tried before, but I couldn't get it to work; it kept escaping `$HOME` in some way or other, and the error that was pointed out above was due to me trying to then switch to `~`. I'll try again tomorrow :)

Reply:
I've kept `--run`: It seems impossible to use `--set` and still point to the user's `$HOME` correctly, as the resulting shell code has quotes around the variable value with `--set`. Otherwise the shell output is identical.